topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsConfiguration.java

```diff
@@ -58,7 +67,34 @@ public void start(Config config, Properties kafkaProperties) {
             String storeTopic = config.get(Config.STORE_TOPIC);
             String storeName = config.get(Config.STORE_NAME);
 
-            long timeoutMillis = config.get(Config.STALE_RESULT_TIMEOUT);
+            // check if entry topic has the right configuration
+            Admin admin = Admin.create(kafkaProperties);
+            DescribeClusterResult clusterResult = admin.describeCluster();
+            int clusterSize = clusterResult.nodes().get().size();
+            Set<String> topics = admin.listTopics().names().get();
+            if (topics.contains(storeTopic)) {
+                TopicDescription topicDescription = admin.describeTopics(Collections.singleton(storeTopic)).values().get(storeTopic).get();
+                int rf = topicDescription.partitions().get(0).replicas().size();
+                ConfigResource isrCR = new ConfigResource(ConfigResource.Type.TOPIC, TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG);
+                DescribeConfigsResult configsResult = admin.describeConfigs(Collections.singleton(isrCR));
+                int minISR = 0;
+                try {
+                    minISR = Integer.parseInt(configsResult.values().get(isrCR).get().get(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG).value());
+                } catch (Exception ignored) {
+                }
+                if (rf != Math.min(3, clusterSize) || minISR != rf - 1) {
+                    log.warn("Durability of the topic is not sufficient for production use - replicationFactor: {}, {}: {}",
+                            rf, TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, minISR);
+                }
+            } else {
+                int rf = Math.min(3, clusterSize);
+                int minISR = rf - 1;
+                NewTopic newTopic = new NewTopic(storeTopic, 1, (short) rf)
+                        .configs(Collections.singletonMap(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, String.valueOf(minISR)));
+                admin.createTopics(Collections.singleton(newTopic)).all().get();
+            }
```

Comment:
the entire logic of this code is kind of blocking because of the usage of the `get` calls on the different `KafkaFuture`.
I was wondering if we need a more async way (so maybe some Vert.x `executeBlocking` calls) to do that but @tombentley knows better than me if it fits well in the overall TO logic calling this piece of code.

Reply:
This is configuration validation, and the topic needs to exist before Kafka Streams kick-in.

Reply:
Yeah, I know but it doesn't mean that the code cannot be async. You can chain async calls in order to do configuration first and then starting Kafka Streams. But as I said I don't know better the calling code and the overall TO architecture, so @tombentley can answer to that. Maybe I am wrong :-)

Reply:
You should be able to use `KafkaFuture.whenComplete()` in a similar way to vertx's `Future.compose()`. This is what `KafkaAvailability` does IIRC. It's also possible to convert `KafkaFuture` to vertx `Future` (see io.strimzi.operator.common.Util#kafkaFutureToVertxFuture, and note that it completes the `Future` on the vertx event loop thread).

Reply:
Do we really want/need to drag Vert.x into this configuration?
Afais, it's 2-3 get() calls, which shouldn't take long.
It would just make this configuration even more complex -- I'm already over the class import limit. :-)

Reply:
This is how our operators are developed ... the code is async everywhere using Vert.x. We wrapped the Kubernetes REST API calls in the same way and of course any kind of call to Kafka as Admin client. I don't think you can make any assumption on the time needed for the get() calls.

Reply:
OK, let me try to impl this config::start in an async way then.